samples/sources_v1alpha2_websocketsource.yaml

```diff
@@ -0,0 +1,49 @@
+# Use the ServiceEntry and VirtualService if your istio installation
+# forbids external communication.
+# 
+# apiVersion: networking.istio.io/v1alpha3
+# kind: ServiceEntry
```

Comment:
So, to get the external connection to run, these Isito APIs (e.g. `ServiceEntry`, `VirtualService`) are needed to be configured.  IMO that's a bit getting in the way of a normal developer, that just wants to have its source connect to outside.

IMO Istio APIs here, e.g. w/ black/white listing and/or IP address ranges are odd.

Is it possible to make this simple, in a future version of Knative Eventing (Sources)? 

e.g. have something in the CRDs/APIs that deal with that stuff - and make it simpler ? 

@vaikas-google @evankanderson @n3wscott @grantr

Reply:
Yes, my experience is that having full control over the hosts contacted by a container is not always possible. I'm trying to develop complex event sources and I've found it difficult to whitelist all servers.

First of all, when using HTTPS to connect to the outside, wildcard domains in ServiceEntry seem not to work properly, you need to specify all exact hostnames.

Second, there might be cloud providers that will do DNS load balacing to serve their clients and who writes the EventSource cannot be aware of all hosts that the cloud provider is going to use to handle requests.

A smarter solution might be to define what's inside the mesh and what's outside. Currently everything ('*') is inside unless otherwise declared.